src/TagHelperPack/HtmlHelperExtensions.cs

```diff
@@ -95,4 +98,58 @@ public static IHtmlContent Editor(this IHtmlHelper htmlHelper, ModelExpression m
 
         return htmlHelper.Editor(expression, templateName, htmlFieldName, additionalViewData);
     }
+
+    /// <summary>
+    /// Merge values from 2 anonymous or IDictionary objects. Values of overlapping keys from the 'existing values' object are replaced by the values 
+    /// from the 'new values' object except if the keys are 'class' or 'style', in which case the values are concatentated with a space or ; respectively.
+    /// </summary>
+    /// <param name="newHtmlAttributesObject">new values</param>
+    /// <param name="existingHtmlAttributesObject">existing values</param>
+    /// <returns></returns>
+    internal static IDictionary<string, object> MergeHtmlAttributes(this IHtmlHelper helper, object newHtmlAttributesObject, object existingHtmlAttributesObject)
```

Comment:
I haven't reviewed this fully but at first glance it seems quite a bit larger than a similar method in the ASP.NET Core itself that does something very similar: https://source.dot.net/#Microsoft.AspNetCore.Mvc.ViewFeatures/DefaultEditorTemplates.cs,192

Could it be simplified?